FC-0068 docs: Documentation Templates (How-to, Reference, Concept)

Apgomeznext opened the pull request:
This PR adds templates for the different document types according to the diataxis used in the Open edX repository.

Review version: https://docsopenedxorg--619.org.readthedocs.build/en/619/documentors/references/doc_templates.html

How To template doc: https://docs.google.com/document/d/1nDCoSnNApQciYD042cqCadk94LzWf4kQSPsGi0VeIBI/edit?tab=t.0#heading=h.fsh8gw4ee3kc
Reference template: https://docs.google.com/document/d/1QHr5sfKeeQ2_PzOMfw53O8QOZHJoxxNyKcuFhQUHCDw/edit?tab=t.0#heading=h.6b7r0b9noosl
Concept Template: https://docs.google.com/document/d/1n3hfDkPl2ftsPreLz601Aa_zh4E6Zqft6d8ZcMCAZK0/edit?tab=t.0#heading=h.6b7r0b9noosl

openedx-webhooks commented:
Thanks for the pull request, @Apgomeznext!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.

<!-- comment:end_of_wip -->


#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/docs-openedx-org-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

jswope00 commented on source/documentors/references/doc_templates.rst:
This seems redundant now that we have source/documentors/references/templates/index.rst. 

The error you are getting seems somewhat related to the fact that this doc_templates.rst file exists, but it is not in any TOC.

jswope00 reviewed (changes requested):
The changes removed a reference to doc_templates.rst file, which is causing a "does not exist in TOC" error. It looks like that file can be deleted because it has been replaced by index.rst.

jswope00 reviewed (approved):
Looks good

jesgreco commented on source/documentors/references/templates/doc_concept_template.rst:
@Apgomeznext there is a punctuation error (last full stop) which should be fixed, and also there is an extra space between brackets:
Lines 6 should be like this: For example, “What is an XBlock? The Open edX platform provides different components, called XBlocks, that can work to create a course, like text, video, assessment, and discussions.” (Replace this text with your content).

jesgreco commented on source/documentors/references/templates/doc_how_to_template.rst:
```diff
@@ -0,0 +1,83 @@
+
+A How-To guide document guides the reader through the steps to solve a real-world problem. This documentation is goal-oriented and similar to recipes, with directions that guide the reader through the steps to achieve a specific end.
+
+For example, following the steps in this How-To guide, you can create a how-to document for any repository for the Open edX platform. (Description)
```
Add a full stop after "(Description)."

jesgreco commented on source/documentors/references/templates/doc_how_to_template.rst:
```diff
@@ -0,0 +1,83 @@
+
+	A How-To guide document guides the reader through the steps to solve a real-world problem. This documentation is goal-oriented and similar to recipes, with directions that guide the reader through the steps to achieve a specific end.
+
+	For example, following the steps in this How-To guide, you can create a how-to document for any repository for the Open edX platform. (Description)
```
Same here with punctuation.

jesgreco commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -0,0 +1,232 @@
+
+1.  Open the document you want to embed. Add ``.. START HERE`` at the beginning and ``.. END HERE`` at the end of the section you want to embed.
+
+2. You will embed the other document in the document. Use the following code to the exact place where you want to add that particular section. Do not forget to edit the document link that you want to include in your Quickstart
```
Add a final stop after "Quickstart".

jesgreco commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -0,0 +1,232 @@
+Useful Resources
+****************
+
+Here is a list of the most used resources to create a Quickstart
```
Add a final stop after "Quickstart".

sarina replied:
```suggestion
Here is a list of the most used resources to create a Quickstart:
```

jesgreco commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -0,0 +1,232 @@
+
+Usually, for this type of document, you can use different directives like ``warning``, ``note``,  and ``see also``, among others.
+
+..  see also:: Review the Style Guide to learn more about directives and other resources to create your documentation.
```
Should we add the link to the Style Guide, right?

sarina replied:
Yes, please link to the Style Guide

sarina replied:
And please add this See Also to all 4 template guides.

jesgreco commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -0,0 +1,232 @@
+
+	1.  Open the document you want to embed. Add ``.. START HERE`` at the beginning and ``.. END HERE`` at the end of the section you want to embed.
+
+	2. You will embed the other document in the document. Use the following code to the exact place where you want to add that particular section. Do not forget to edit the document link that you want to include in your Quickstart
```
Final stop after "Quickstart".

jesgreco commented on source/documentors/references/templates/doc_reference_template.rst:
```diff
@@ -0,0 +1,136 @@
+
+Reference material is information-oriented. Regarding software, reference guides describe the software - APIs, classes, functions, etc. In the context of the Open edX® Docs, reference docs are often in the form of informational lists (e.g., Resources for Students) and tables (e.g., Course Access Roles and Privileges).
+
+Reference material is like a map. A map tells you what you need to know about the territory without having to go out and check it for yourself; a reference guide serves the same purpose for the product and its internal machinery. For example, this part works as the introduction to the article and as a description of the product of what this document is about. (Replace this text with your content. ...
```
Full stop should be after the parenthesis:  (Replace this text with your content).

sarina replied:
```suggestion
Reference material is like a map. A map tells you what you need to know about the territory without having to go out and check it for yourself; a reference guide serves the same purpose for the product and its internal machinery. For example, this part works as the introduction to the article and as a description of the product of what this document is about. 
```

I think "(Replace this text with your content)" is more confusing here than helpful.

jesgreco commented on source/documentors/references/templates/doc_reference_template.rst:
```diff
@@ -0,0 +1,136 @@
+	Reference material is like a map. A map tells you what you need to know about the territory without having to go out and check it for yourself; a reference guide serves the same purpose for the product and its internal machinery. For example, this part works as the introduction to the article and as a description of the product of what this document is about. (Replace this text with your content ...
+
+	The purpose of this document is to be as straightforward as possible. In addition to an accurate and precise description of the product you’re referencing, use examples of the uses and functions of the product to make it more comprehensive.
+	For example, list details or a product glossary that the reader could need to use. (Replace this text with your content.)
```
Full stop should be after the parenthesis: (Replace this text with your content).

jesgreco reviewed (commented):
I've left some comments mostly related to punctuation to be fixed :)

sarina commented on source/documentors/references/templates/doc_concept_template.rst:
```diff
@@ -0,0 +1,66 @@
+
+A concept is a document that clarifies and illuminates a particular topic. It is understanding-oriented. It could be considered a conceptual guide. This type of document answers a why question. It could be descriptive, historical, or even propose different alternatives to explain the bigger picture and give context. 
+
+For example, “What is an XBlock? The Open edX platform provides different components, called XBlocks, that can work to create a course, like text, video, assessment, and discussions.”  (Replace this text with your content.)
```
```suggestion
For example, “What is an XBlock? The Open edX platform provides different components, called XBlocks, that can work to create a course, like text, video, assessment, and discussions.”  (Replace this text with your content).
```

sarina commented on source/documentors/references/templates/doc_concept_template.rst:
```diff
@@ -0,0 +1,66 @@
+
+For example, “What is an XBlock? The Open edX platform provides different components, called XBlocks, that can work to create a course, like text, video, assessment, and discussions.”  (Replace this text with your content.)
+
+Since this type of document seeks to explain a topic, you will likely need to add images to exemplify the topic you are talking about. For example:
```
```suggestion
Since this type of document seeks to explain a topic, you will likely need to add images to illustrate the topic you are talking about. For example:
```

"exemplify" is a more complicated english word than "illustrate"

sarina commented on source/documentors/references/templates/index.rst:
```diff
@@ -0,0 +1,11 @@
+Open edX Documents Templates
```
```suggestion
Open edX Document Templates
```

sarina commented on source/documentors/references/templates/doc_concept_template.rst:
```diff
@@ -0,0 +1,66 @@
+Since this type of document seeks to explain a topic, you will likely need to add images to exemplify the topic you are talking about. For example:
+
+
+.. image:: /_images/documentors_howto/image_example.png
```
Why is the example text above about XBlocks, but this image unrelated to XBlocks? It's confusing. I'd make the image and the example text above at line 6 match.

sarina commented on source/documentors/references/templates/doc_concept_template.rst:
```diff
@@ -0,0 +1,66 @@
+External Links
+--------------
+
+You can have `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ and `indirect links`_, that are useful if you want to link to the same thing multiple times, or if the URL is really long and you want things to read more cleanly.
```
```suggestion
You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

sarina commented on source/documentors/references/templates/doc_reference_template.rst:
```diff
@@ -0,0 +1,136 @@
+Reference material is like a map. A map tells you what you need to know about the territory without having to go out and check it for yourself; a reference guide serves the same purpose for the product and its internal machinery. For example, this part works as the introduction to the article and as a description of the product of what this document is about. (Replace this text with your content. ...
+
+The purpose of this document is to be as straightforward as possible. In addition to an accurate and precise description of the product you’re referencing, use examples of the uses and functions of the product to make it more comprehensive.
+For example, list details or a product glossary that the reader could need to use. (Replace this text with your content.)
```
```suggestion
For example, list details or a product glossary that the reader could need to use.
```
I think "(Replace this text with your content)" is more confusing here than helpful.

sarina commented on source/documentors/references/templates/doc_reference_template.rst:
```diff
@@ -0,0 +1,136 @@
+External Links
+--------------
+
+You can have `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ and `indirect links`_, that are useful if you want to link to the same thing multiple times, or if the URL is really long and you want things to read more cleanly.
```
```suggestion
You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

sarina commented on source/documentors/references/templates/doc_reference_template.rst:
```diff
@@ -0,0 +1,136 @@
+	External Links
+	--------------
+
+	You can have `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ and `indirect links`_, that are useful if you want to link to the same thing multiple times, or if the URL is really long and you want things to read more cleanly.
```
```suggestion
	You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

sarina commented on source/documentors/references/templates/doc_concept_template.rst:
```diff
@@ -0,0 +1,66 @@
+	External Links
+	--------------
+
+	You can have `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ and `indirect links`_, that are useful if you want to link to the same thing multiple times, or if the URL is really long and you want things to read more cleanly.
```
```suggestion
	You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

sarina commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -0,0 +1,232 @@
+External Links
+--------------
+
+You can have `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ and `indirect links`_, that are useful if you want to link to the same thing multiple times, or if the URL is really long and you want things to read more cleanly.
```
```suggestion
You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

sarina commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -0,0 +1,232 @@
+	External Links
+	--------------
+
+	You can have `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ and `indirect links`_, that are useful if you want to link to the same thing multiple times, or if the URL is really long and you want things to read more cleanly.
```
```suggestion
	You can use `inline links <https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#linking>`_ or `indirect links`_. Indirect links are useful if you want to link to the same thing multiple times, or if the URL is really long and you want the raw RST to read more cleanly.
```

sarina commented on source/documentors/references/templates/doc_how_to_template.rst:
```diff
@@ -0,0 +1,83 @@
+
+7. Add any reference that could be important for someone to understand the topic. For example, here is additional information about `how to style for Read the Docs <https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/>`_.
+
+Now, you have a how-to document. ¡Congratulations! 
```
```suggestion
Now, you have a how-to document. Congratulations! 
```

sarina commented on source/documentors/references/templates/doc_how_to_template.rst:
```diff
@@ -0,0 +1,83 @@
+
+	7. Add any reference that could be important for someone to understand the topic. For example, here is additional information about `how to style for Read the Docs <https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/>`_.
+
+	Now, you have a how-to document. ¡Congratulations! 
```
```suggestion
	Now, you have a how-to document. Congratulations! 
```

sarina commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -0,0 +1,232 @@
+Quickstrat Document Template
```
```suggestion
Quickstart Document Template
```
Be sure you're using an English spellchecker on these documents

sarina commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -0,0 +1,232 @@
+
+#. **Common Next Steps** – Information on where to go next, often with links to detailed documentation.
+
+A good quickstart is designed for users who want to dive in with minimal setup. It benefits developers, allowing them to try out a tool or platform without wading through extensive documentation upfront. 
```
```suggestion
A good quickstart is designed for users who want to dive in with minimal setup. For example, allowing developers to try out a tool or platform without wading through extensive documentation upfront, or empowering course authors to quickly get up and running with a complex area of Studio.
```

To call out that Quickstarts are not _just_ for developers, which is how this currently reads. Quickstarts are for all personas.

sarina commented on source/documentors/references/templates/doc_concept_template.rst:
```diff
@@ -3,15 +3,15 @@ Concept Document Template
 A concept is a document that clarifies and illuminates a particular topic. It is understanding-oriented. It could be considered a conceptual guide. This type of document answers a why question. It could be descriptive, historical, or even propose different alternatives to explain the bigger picture and give context. 
 
-For example, “What is an XBlock? The Open edX platform provides different components, called XBlocks, that can work to create a course, like text, video, assessment, and discussions.”  (Replace this text with your content).
+For example, “What is an XBlock? The Open edX platform provides different components, called XBlocks, that can work to create a course, like text, video, assessment, and discussions”.
```
My comment about the image example was that this line ("What is an XBlock") might be clearer if the associated image _also_ was about the same thing. So you changed the image to one about Superset, which is still unrelated to this line about XBlocks. It's not a big deal but that was what I was saying.

Apgomeznext replied:
You are right. Sorry about the confusion.

sarina commented on source/documentors/references/templates/doc_quickstart_template.rst:
```diff
@@ -109,7 +112,7 @@ Directives:
 Usually, for this type of document, you can use different directives like ``warning``, ``note``,  and ``see also``, among others.
 
-..  see also:: Review the Style Guide to learn more about directives and other resources to create your documentation.
+..  see also:: Review the `Style Guide <https://docs.openedx.org/en/latest/documentors/references/doc_style_guide.html>`_ to learn more about directives and other resources to create your documentation.
```
This isn't rendering in the output

<img width="811" alt="image" src="https://github.com/user-attachments/assets/ffe6c8ee-b951-4a1d-9e22-0d7ab41687ea">

https://docsopenedxorg--619.org.readthedocs.build/en/619/documentors/references/templates/doc_quickstart_template.html

sarina commented on source/documentors/references/templates/doc_reference_template.rst:
```diff
@@ -66,17 +66,18 @@ folder.
 
 By default, it will use the title of the doc as the link text but you can override that with :doc:`other text <doc_how_to_template>` if you want.
 
+..  see also:: Review the `Style Guide <https://docs.openedx.org/en/latest/documentors/references/doc_style_guide.html>`_ to learn more about directives and other resources to create your documentation.
```
not rendering
<img width="388" alt="image" src="https://github.com/user-attachments/assets/71bcbd7f-3b2f-4bbf-be36-4af0245e8eb0">

Apgomeznext commented:
The only thing that needs to be added to these documents is the link to the style guide. The error that this PR showed this morning was because I had added the link as it should have once it was published, but I deleted it to avoid the error.

sarina commented on source/documentors/references/templates/doc_how_to_template.rst:
```diff
@@ -0,0 +1,87 @@
+
+  <h1 class="title">Title</h1>
+
+.. seealso:: To learn more about how to add a code, visit the `Code blocks documentation <https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#code-blocks>`_.
```
```suggestion
.. seealso:: To learn more about how to add a code block, visit the `Code blocks documentation <https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#code-blocks>`_.
```

sarina commented on source/documentors/references/templates/doc_how_to_template.rst:
```diff
@@ -0,0 +1,87 @@
+  		<h1 class="title">Title</h1>
+
+
+	.. seealso:: To learn more about how to add a code, visit the `Code blocks documentation <https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#code-blocks>`_.
```
```suggestion
	.. seealso:: To learn more about how to add a code block, visit the `Code blocks documentation <https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#code-blocks>`_.
```
